### DISCUSSION TITLE
StaticTextBlock Localizer/Language

### DISCUSSION THREAD

**Discussion by ZioZenigata** [serenity-is/Serenity#6387] [Help] [answered]
Hi, 
Is there any way to show StaticTextBlock in language?
I use this snipped code in myForm.cs

[StaticTextBlock(HideLabel = true, IsHtml = true, IsLocalText = true, Text = "HELLO"), Localizable(true)]
public string WelcomeMessage { get; set; }

But in my Administration/Translation page there isn't WelcomeMessage entry.
Where I am wrong?
Thanks 
Roberto

**Comment by wezmag** [comment, marked as answer]
Hello Roberto,

You need to add a `LocalText` in `Texts.cs` file.
For example:
``` csharp
[NestedLocalTexts]
public static partial class Texts
{
   public static class YourModuleName
   {
      public static LocalText WelcomeMessage = "Your welcome message";
   }
}
```

And then in `appsettings.json`, add `YourModuleName` in `LocalTextPackages.Site`

``` json
"LocalTextPackages": {
  "Site": "^(Controls|Db|Dialogs|Enums|Forms|Permission|Site|Validation|YourModuleName)\\.",
  "Login": "^(Forms\\.Membership\\.Login|Db\\.Administration\\.User|Validation\\.Required|Dialogs)\\."
},
```

finally, in your WelcomeMessage property, change `Text = "HELLO"` to `Text = "YourModuleName.WelcomeMessage"`.

**Tested with SERENE version 5.2.4.0**

**Reply by wezmag** [reply]
Do not forget to add translation of the message in Translation module.

**Reply by ZioZenigata** [reply]
Thanks a lot!
